libs/api/domains/form-system/src/lib/groups/groups.service.ts

```diff
@@ -0,0 +1,107 @@
+import { Inject, Injectable } from '@nestjs/common'
+import { LOGGER_PROVIDER, Logger } from '@island.is/logging'
+import { ApolloError } from '@apollo/client'
+import { AuthMiddleware, User } from '@island.is/auth-nest-tools'
+import {
+  GroupsApi,
+  ApiGroupsGroupIdGetRequest,
+  ApiGroupsPostRequest,
+  ApiGroupsGroupIdDeleteRequest,
+  ApiGroupsGroupIdPutRequest,
+  GroupCreationDto,
+  GroupUpdateDto,
+} from '@island.is/clients/form-system'
+import {
+  GetGroupInput,
+  CreateGroupInput,
+  DeleteGroupInput,
+  UpdateGroupInput,
+} from '../../dto/groups.input'
+import { Group } from '../../models/group.model'
+
+@Injectable()
+export class GroupsService {
+  constructor(
+    @Inject(LOGGER_PROVIDER)
+    private logger: Logger,
+    private formsApi: GroupsApi,
+  ) {}
+
+  handleError(error: any, errorDetail?: string): ApolloError | null {
+    const err = {
+      error: JSON.stringify(error),
+      category: 'groups-service',
+    }
+    this.logger.error(errorDetail || 'Error in groups service', err)
+
+    throw new ApolloError(error.message)
+  }
+
+  private handle4xx(error: any, errorDetail?: string): ApolloError | null {
+    if (error.status === 403 || error.status === 404) {
+      return null
+    }
+    return this.handleError(error, errorDetail)
+  }
```

Comment:
I created a shared errorHandler utility under utils/errorHandler.ts

Reply:
You should also make the `handleError` function a utility instead of copying it to only change the category string. You can resolve the fallback value for  `errorDetail` by doing something like 
```
handleError(error: any, category: string, errorDetail?: string): ApolloError | null {
  const err = {
    error: JSON.stringify(error),
    category,
  }
  this.logger.error(errorDetail || `Error in ${category}`, err)

  throw new ApolloError(error.message)
}
```